apple/DemoApp/Demo/DemoModel.swift

```diff
@@ -140,13 +140,17 @@ extension DemoModel {
         return .idle
     }
 
+    func handleError(_ error: Error) {
+        errorMessage = error.localizedDescription
+        appState = .idle
+    }
```

Comment:
Something like this may be useful _if_ this handle error function is potentially valuable during navigation as well. Ok if it's not.

```suggestion
    func handleError(_ error: Error, newAppState: DemoAppState = .idle) {
        errorMessage = error.localizedDescription
        appState = newAppState
    }
```